File: src/main/java/io/github/tanguygab/conditionalactions/conditions/ConditionalOutput.java
package io.github.tanguygab.conditionalactions.conditions;

import org.bukkit.OfflinePlayer;
import org.bukkit.configuration.ConfigurationSection;

import java.util.LinkedHashMap;
import java.util.Map;

public class ConditionalOutput {

    private final Map<ConditionGroup,String> outputs = new LinkedHashMap<>();

    public ConditionalOutput(ConditionManager manager, ConfigurationSection section) {
        section.getValues(false).forEach((output,condition)-> outputs.put(manager.getCondition(String.valueOf(condition)), output));
    }

    public String getOutput(OfflinePlayer player) {
        for (ConditionGroup condition : outputs.keySet())
            if (condition.isMet(player))
                return outputs.get(condition);
        return "";
    }

}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/conditions/ConditionalOutput.kt
package io.github.tanguygab.conditionalactions.conditions

import org.bukkit.OfflinePlayer
import org.bukkit.configuration.ConfigurationSection

class ConditionalOutput(manager: ConditionManager, section: ConfigurationSection) {
    private val outputs = mutableMapOf<ConditionGroup, String>()

    init {
        section.getValues(false).forEach { (output: String, condition: Any) ->
            outputs[manager.getCondition(condition.toString())] = output
        }
    }

    fun getOutput(player: OfflinePlayer?): String? {
        for (condition in outputs.keys)
            if (condition.isMet(player))
                return outputs[condition]
        return ""
    }
}
